gateway/api_definition.go

```diff
@@ -409,9 +409,21 @@ func (f *nestedApiDefinitionList) set(defs []*apidef.APIDefinition) {
 	}
 }
 
+func (f *nestedApiDefinitionList) all() []nestedApiDefinition {
+	result := make([]nestedApiDefinition, 0, len(f.Message))
+	for _, v := range f.Message {
+		// Clear nil APIDefinitions avoiding nil panics.
+		if v.APIDefinition == nil {
```

Comment:
We discussed this previously. Why would an API definition comes as nil? We should handle that part. In here if we want to do something, logging that api definition is invalid instead of returning valid ones.

Reply:
I agree, returning an error from such an API call (FromRPC, FromDashboard) makes sense, rather than panic (and instead of just skipping the nil value). I'll extend the PR accordingly. I don't think just logging a skipped nil is enough.

The error is also triggered from tests (`gateway.TestSyncAPISpecsRPCFailure_CheckGlobals`). If MDCB triggers a panic, which it did before and is logged in the JIRA ticket, that has it's own it's own owners who can investigate further. At least one case was that incompatible versions of MDCB and GW have been run.

Reply:
👍